Rework message-handling methods in InputStream and Consumer.

Evildoor opened the pull request:
Both InputStream and Consumer perform a similar sequence of actions with messages. However, it is done in a contradictory and confusing manner: in Consumer `next()` is called from the "upper" levels and calls other methods that perform actual work, while in the InputStream `next()` is a method that performs work and is called by `get_message()` which, in turn, is called from "above". This will be resolved by standardizing the methods in the following way (each method calls the previous one):

1. `get_raw_item()`: get an item from "lower" level.
2. `get_item()`: process the item.
3. `next()`: return the item.

~~Consumer has to be changed in this regard as well - however, without batch processing implemented, it would be simply a matter of changing some names and adding `get_item()` that does nothing apart from `return get_raw_item()`. Therefore, I decided to do this later, together with batch processing (out of discussion of which this whole idea arose).~~

mgolosova reviewed (changes requested):
The suggested commit is a bit difficult to read, please split it somehow... for example:
* move "get_raw_item" functionality into a separate method";
* switch "get_message"/"next" methods;
* rename "get_message" into "get_item".

The PR mentions both `InputStream` and `Consumer` classes, so it's better to bring the repo to the point where these two are consistent and obey same logic. So, despite the triviality of changes on the `Consumer` side, please do the second part here as well. Or at least reword the commit(s?) and PR description so that one didn't end up with a feeling that it's just a part of something bigger, and that this "bigger" was left unfinished for whatever reason ;)

You decided not to split `<message>=<raw_data><EOM>` into two raw items (`<data>` and `<EOM>`) here, right? Is it planned to be done later, or will be left as-is (which is fine, but, as it was discussed in the batch processing proposal, leads to a mixed logic in the future -- while the PR states that it's fighting against "confusing" things, so...)
It can as well be done in a separate PR, but my point here is that "bringing order into message-reading part" includes all these changes, aimed to formalize "items", their usage and handling -- so it all could be done in the same PR. Or, again, the general issue of "contradictory and confusing manner" shouldn't be stated as the (single and standalone) PR's motivation, since it's not fully solved here...

---

Changes requested:
* split the commit;
* reword the PR description ***OR*** include all the changes that the current one implies;
* reword method's descriptions (see suggestions below, mb reword them):
  * "message:" -> "raw/stream item",
  * what should be taken as "an item" in each case,
  * "convert" -> "get" or "produce", etc.

Evildoor commented on Utils/Dataflow/pyDKB/dataflow/communication/stream/InputStream.py:
```diff
@@ -175,3 +173,17 @@ def next(self):
+    def next(self):
+        """ Get next message from the input stream.
+
+        :raises: StopIteration: no messages left
+        :returns: parsed next message,
```
Reword description.

Evildoor commented:
> The suggested commit is a bit difficult to read, please split it somehow... for example:
> 
> * move "get_raw_item" functionality into a separate method";
> * switch "get_message"/"next" methods;
> * rename "get_message" into "get_item".

Done.

> The PR mentions both `InputStream` and `Consumer` classes, so it's better to bring the repo to the point where these two are consistent and obey same logic. So, despite the triviality of changes on the `Consumer` side, please do the second part here as well. Or at least reword the commit(s?) and PR description so that one didn't end up with a feeling that it's just a part of something bigger, and that this "bigger" was left unfinished for whatever reason ;)

Fair enough, updated `Consumer`.

> You decided not to split `<message>=<raw_data><EOM>` into two raw items (`<data>` and `<EOM>`) here, right? Is it planned to be done later, or will be left as-is (which is fine, but, as it was discussed in the batch processing proposal, leads to a mixed logic in the future -- while the PR states that it's fighting against "confusing" things, so...)
> It can as well be done in a separate PR, but my point here is that "bringing order into message-reading part" includes all these changes, aimed to formalize "items", their usage and handling -- so it all could be done in the same PR. Or, again, the general issue of "contradictory and confusing manner" shouldn't be stated as the (single and standalone) PR's motivation, since it's not fully solved here...

Yes, the logic of markers and messages will be changed elsewhere, be it a separate PR or PR about batch processing. Both terms - "contradictory and confusing manner" and "bringing order" intended to combat it - are rather broad, but the idea of this PR was to deal specifically with two different directions of method hierarchy. Updated PR description to better reflect this.

> Changes requested:
> ...
> * reword method's descriptions (see suggestions below, mb reword them):
>   
>   * "message:" -> "raw/stream item",
>   * what should be taken as "an item" in each case,
>   * "convert" -> "get" or "produce", etc.

Done.

mgolosova reviewed (changes requested):
Another set of suggestions for docstrings is below.
*Note: some of them may conflict (for they change very close lines), so you will have to solve the conflict in terminal.*

Some changes fell out of their thematic commit and end up in unrelated one (see in the comments below); please fix it.

When you're done with the docstrings and commits reordering, you will also need to update the `pyDKB` version to finish the PR *(check the how-to https://github.com/PanDAWMS/dkb/wiki/How-to%3A-versioning)*.

The functional part looks fine for me, thank you!

---

Unrequired, but I believe worth mentioning: **it'd be nice to specify a *scope* of your changes in the commit summaries.**

***What is scope***
*Scope* is a (part of) the commit summary prefix, that contains additional contextual information about the commit; how to define the set of existing *scopes* is up to the developers, but one of the variants is to use it to specify which part of the codebase has changed.

[Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/), for example, suggest to specify scope as a part of the prefix (`<type>[optional scope]: <description>`); in this repository, as you might have noticed, we do not follow this convention -- but I very often add the scope prefix to commits.

***Why to use***
Please take a look at the PR's commit history (https://github.com/PanDAWMS/dkb/pull/430/commits) and, for example, this page: https://github.com/PanDAWMS/dkb/tree/61517365ad95e1b161e106ac3c7605bb9f282218/Utils

From the commit summaries seen there it's not very clear what *exactly* happened here or there; some methods and technical terms are mentioned, but unless one dives into the commit's diff, it's impossible to get the idea and realize if this commit is significant for something being investigated right here and now. *(Extended description might help, but there's none of that.)* It would be fine to describe a change this way if it was the library-wide or, better, the repository-wide (e.g. if we suddenly decide to avoid word "message" and use instead something else, e.g. "statement") -- for one doesn't need to know the code details to understand the effect of this change. Or if it was a really, *really* unsignificant change that affects almost nothing (like a typo fix) -- well, it doesn't matter where exactly it was done: a typo is a typo, in any context.

Should one speak about, e.g. d0a680b's changes, (s)he would describe it somehow like this: "Separate getting the raw item acton into an individual method from method \`next\` in pyDKB.dataflow.communication.stream.InputStream class." Well, maybe not with the full class name: "...in \`InputStream\` class of the \`pyDKB\` library (\`communication\` submodule)." This statement fully describes the change -- what and where happened; any additional details may be needed only to explain why and what for, but if it's obvious -- may as well be omitted. *Where*, the *scope*, is an important part of this statement, for without the one the statemeint might be even incorrect. To be correct, and to be easily understandable, it requires both (the speaker and the listener) to be in exactly same context... which is not even the `communication` submodule of the `pyDKB` library, but a specific class of this submodule. And this context switches even within this single PR (so that even the reviewer/author pair can not in the same context for the entire PR), not to mention the whole repository's commit history (where the reader may know *very* little about the `pyDKB` package content, even if uses it in stages).

Of course, in the commit summary we have only 50 symbols of the recommended length (and 72 at most); so obviously it must be shortened. So -- the *scope* as a prefix, like `pyDKB/communication/ImputStream`, or `pyDKB/communication` if the `ImputStream` is mentioned in the description. But completely throw it away... it can be reconstructed, of course -- by checking what files has changed or so. But what I'm saying here is that it's not nice to make others to do this kind of digging for every commit ;)

Evildoor commented on Utils/Dataflow/pyDKB/dataflow/communication/consumer/Consumer.py:
```diff
@@ -111,12 +111,23 @@ def get_message(self):
         return msg
 
+    def get_item(self):
+        """ Get next stream item (constructed of raw items).
```
Agreed on "stream"->"processing", but "raw" looks fine to me as-is. This method constructs processing item from raw items, while `get_raw_item()` would explain what exactly this "raw" item is.

Evildoor replied:
Done except the "raw" part mentioned above.

Evildoor commented:
Several force-pushes to properly organize commits and add scopes.

Evildoor commented on Utils/Dataflow/pyDKB/dataflow/communication/stream/InputStream.py:
```diff
@@ -138,31 +138,29 @@ def parse_message(self, message):
+        return self.__iterator.next()
+
+    def get_item(self):
+        """ Get next stream item (constructed of raw items).
```
Done.

Evildoor commented on Utils/Dataflow/pyDKB/dataflow/communication/consumer/Consumer.py:
```diff
@@ -99,8 +99,8 @@ def get_source_info(self):
-    def get_message(self):
-        """ Get new message from current source.
+    def get_raw_item(self):
+        """ Get new raw item (message) from current source.
```
Done.

Evildoor commented on Utils/Dataflow/pyDKB/dataflow/communication/consumer/Consumer.py:
```diff
@@ -111,12 +111,23 @@ def get_message(self):
+        return self.get_raw_item()
+
     def next(self):
         """ Return new Message, read from input stream. """
```
Done.

Evildoor commented on Utils/Dataflow/pyDKB/dataflow/communication/stream/InputStream.py:
```diff
@@ -149,28 +149,18 @@ def get_raw_item(self):
+        """ Get next stream item (constructed of raw items).
 
-        :returns: parsed next message,
+        :returns: parsed next item,
```
True, separated them.

Evildoor commented on Utils/Dataflow/pyDKB/dataflow/communication/stream/InputStream.py:
```diff
@@ -138,31 +138,29 @@ def parse_message(self, message):
-    def get_message(self):
-        """ Get next message from the input stream.
+    def get_raw_item(self):
+        """ Get next raw item (sequence of symbols) from input.
```
Done.

Evildoor commented on Utils/Dataflow/pyDKB/dataflow/communication/stream/InputStream.py:
```diff
@@ -138,31 +138,29 @@ def parse_message(self, message):
-    def get_message(self):
-        """ Get next message from the input stream.
+    def get_raw_item(self):
+        """ Get next raw item (sequence of symbols) from input.
```
Done.

Evildoor commented:
@mgolosova, suggestions were answered, scopes added, version updated. Therefore, unless I'm forgetting something, the PR is finally ready to be reviewed again.

mgolosova commented:
@Evildoor, everything looks alright, so -- approved and merged.
Thank you!
